storage/buckets/main.go

```diff
@@ -197,3 +198,71 @@ func removeUser(c *storage.Client, bucketName string) error {
 	// [END remove_bucket_iam_member]
 	return nil
 }
+
+func enableRequesterPays(c *storage.Client, bucketName string) error {
+	ctx := context.Background()
+
+	// [START enable_requester_pays]
+	bucket := c.Bucket(bucketName)
+	if _, err := bucket.Update(ctx, storage.BucketAttrsToUpdate{
+		RequesterPays: true,
+	}); err != nil {
+		return err
+	}
+	// [END enable_requester_pays]
+	return nil
+}
+
+func disableRequesterPays(c *storage.Client, bucketName string) error {
+	ctx := context.Background()
+
+	// [START disable_requester_pays]
+	bucket := c.Bucket(bucketName)
+	if _, err := bucket.Update(ctx, storage.BucketAttrsToUpdate{
+		RequesterPays: false,
+	}); err != nil {
+		return err
+	}
+	// [END disable_requester_pays]
+	return nil
+}
+
+func checkRequesterPays(c *storage.Client, bucketName string) error {
+	ctx := context.Background()
+
+	// [START get_requester_pays_status]
+	attrs, err := c.Bucket(bucketName).Attrs(ctx)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Is requester pays enabled? %v\n", attrs.RequesterPays)
+	// [END get_requester_pays_status]
+	return nil
+}
+
+func downloadUsingRequesterPays(client *storage.Client, object, bucketName, localpath, billingProjectID string) error {
+	ctx := context.Background()
+	// [START storage_download_file_requester_pays]
+	bucket := client.Bucket(bucketName).UserProject(billingProjectID)
+	src := bucket.Object(object)
```

Comment:
Just a thought - maybe moving the UserProject call to the line below, or rename "bucket" to "billedBucket"

```
bucket := client.Bucket(bucketName)
src := bucket.UserProject(billingProjectID).Object(object)
```

Reply:
@broady  The sample makes sense as the billing isn't paid for by the owner of the bucket, but by the requester downloading an object from the bucket.

Reply:
I'm aware.

My suggestion is to move the intent closer to where it actually makes a difference: when you act upon an object. Otherwise the fact that it's a bucket with some other hidden property isn't obvious.